MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,340 +30,302 @@ namespace MediaBrowser.MediaEncoding.Encoder
-            }
-
-            if (string.IsNullOrWhiteSpace(newPaths.Item1))
-            {
-                throw new ResourceNotFoundException("ffmpeg not found");
-            }
-            if (string.IsNullOrWhiteSpace(newPaths.Item2))
-            {
-                throw new ResourceNotFoundException("ffprobe not found");
-            }
-
-            path = newPaths.Item1;
-
-            if (!ValidateVersion(path, true))
-            {
-                throw new ResourceNotFoundException("ffmpeg version 3.0 or greater is required.");
             }
 
-            var config = GetEncodingOptions();
-            config.EncoderAppPath = path;
-            ConfigurationManager.SaveConfiguration("encoding", config);
-
-            Init();
-        }
-
-        private bool ValidateVersion(string path, bool logOutput)
-        {
-            return new EncoderValidator(_logger, _processFactory).ValidateVersion(path, logOutput);
+            return rc;
         }
 
-        private void ConfigureEncoderPaths()
+        private string GetEncoderPath(string path)
         {
-            if (_hasExternalEncoder)
-            {
-                return;
-            }
-
-            var appPath = GetEncodingOptions().EncoderAppPath;
-
-            if (string.IsNullOrWhiteSpace(appPath))
+            if (Directory.Exists(path))
             {
-                appPath = Path.Combine(ConfigurationManager.ApplicationPaths.ProgramDataPath, "ffmpeg");
+                return GetEncoderPathFromDirectory(path, "ffmpeg");
             }
 
-            var newPaths = GetEncoderPaths(appPath);
-            if (string.IsNullOrWhiteSpace(newPaths.Item1) || string.IsNullOrWhiteSpace(newPaths.Item2) || IsSystemInstalledPath(appPath))
+            if (File.Exists(path))
             {
-                newPaths = TestForInstalledVersions();
+                return path;
             }
 
-            if (!string.IsNullOrWhiteSpace(newPaths.Item1) && !string.IsNullOrWhiteSpace(newPaths.Item2))
-            {
-                FFMpegPath = newPaths.Item1;
-                FFProbePath = newPaths.Item2;
-            }
-
-            LogPaths();
+            return null;
         }
 
-        private Tuple<string, string> GetEncoderPaths(string configuredPath)
+        private string GetEncoderPathFromDirectory(string path, string filename)
         {
-            var appPath = configuredPath;
-
-            if (!string.IsNullOrWhiteSpace(appPath))
+            try
             {
-                if (Directory.Exists(appPath))
-                {
-                    return GetPathsFromDirectory(appPath);
-                }
+                var files = FileSystem.GetFilePaths(path);
 
-                if (File.Exists(appPath))
-                {
-                    return new Tuple<string, string>(appPath, GetProbePathFromEncoderPath(appPath));
-                }
-            }
+                var excludeExtensions = new[] { ".c" };
```

Comment:
Why?

Reply:
tbh, no idea. Probably because .c would be found before .exe.